In [1]:
import pandas as pd
from sklearn import preprocessing
from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import accuracy_score
from sklearn.metrics import confusion_matrix

In [2]:
df = pd.read_csv("train.csv")
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

In [3]:
df.isnull()

PassengerId  Survived  Pclass   Name    Sex    Age  SibSp  Parch  Ticket  \
0          False     False   False  False  False  False  False  False   False   
1          False     False   False  False  False  False  False  False   False   
2          False     False   False  False  False  False  False  False   False   
3          False     False   False  False  False  False  False  False   False   
4          False     False   False  False  False  False  False  False   False   
..           ...       ...     ...    ...    ...    ...    ...    ...     ...   
884        False     False   False  False  False  False  False  False   False   
885        False     False   False  False  False  False  False  False   False   
886        False     False   False  False  False  False  False  False   False   
887        False     False   False  False  False  False  False  False   False   
888        False     False   False  False  False  False  False  False   False   

      Fare  Cabin  Embarked  
0    False   True     False  
1    False  False     False  
2    False   True     False  
3    False  False     False  
4    False   True     False  
..     ...    ...       ...  
884  False   True     False  
885  False  False     False  
886  False   True     False  
887  False  False     False  
888  False   True     False  

[889 rows x 12 columns]

In [4]:
df.dropna()

PassengerId  Survived  Pclass  \
1              2         1       1   
3              4         1       1   
6              7         0       1   
10            11         1       3   
11            12         1       1   
..           ...       ...     ...   
869          872         1       1   
870          873         0       1   
877          880         1       1   
885          888         1       1   
887          890         1       1   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
6                              McCarthy, Mr. Timothy J    male  54.0      0   
10                     Sandstrom, Miss. Marguerite Rut  female   4.0      1   
11                            Bonnell, Miss. Elizabeth  female  58.0      0   
..                                                 ...     ...   ...    ...   
869   Beckwith, Mrs. Richard Leonard (Sallie Monypeny)  female  47.0      1   
870                           Carlsson, Mr. Frans Olof    male  33.0      0   
877      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0      0   
885                       Graham, Miss. Margaret Edith  female  19.0      0   
887                              Behr, Mr. Karl Howell    male  26.0      0   

     Parch    Ticket     Fare        Cabin Embarked  
1        0  PC 17599  71.2833          C85        C  
3        0    113803  53.1000         C123        S  
6        0     17463  51.8625          E46        S  
10       1   PP 9549  16.7000           G6        S  
11       0    113783  26.5500         C103        S  
..     ...       ...      ...          ...      ...  
869      1     11751  52.5542          D35        S  
870      0       695   5.0000  B51 B53 B55        S  
877      1     11767  83.1583          C50        C  
885      0    112053  30.0000          B42        S  
887      0    111369  30.0000         C148        C  

[202 rows x 12 columns]

In [5]:
le = preprocessing.LabelEncoder()
le.fit(df["Sex"])
df["Sex"] = le.transform(df["Sex"])

In [6]:
le.fit(df["Embarked"])
df["Embarked"] = le.transform(df["Embarked"])

In [7]:
y = df["Pclass"]
X = df.drop(['Pclass','Cabin','Ticket','Name','PassengerId'],axis=1)

In [8]:
X_train,X_test,y_train,y_test = train_test_split(X,y,test_size=0.3,random_state=0)

In [9]:
from sklearn.naive_bayes import *
clf = BernoulliNB()
ypred = clf.fit(X_train,y_train).predict(X_test)

In [10]:
accuracy_score(y_test,ypred,normalize=True)

0.5917602996254682

# 59.2% Accuracy classified

In [11]:
confusion_matrix(y_test,ypred)

array([[ 30,   8,  32],
       [  9,  10,  30],
       [ 24,   6, 118]], dtype=int64)

# From Confusion Matrix

Correct Prediction - 30+10+118 = 158 ================== 59.17% Accurate

Incorrect Prediction - 9+24+6+8+30+32 = 109 =========== 40.82% Not Accurate

Total = 267